### PR TITLE
API: add v1

### PR DESCRIPTION
**Pull request by ffromani**
Add v1 API. The v1 API is v1alpha1 with the deprecated `DisablePodsFingerprinting` removed, and` NodeGroupConfig` cleaned up accordingly.

We add preparation for webhook, including conversion functions, but this PR does NOT wire them, intentionally.
Deploying conversion webhooks would require changes in the CSV, which would in turn break the upgrade flow (upgrades will still be possible, but they will require manual fixes), and would make the backport even harder.

The consequence of no conversion hooks is the conversion is unidirectional v1alpha1 -> v1.
The operator will **accept** both v1alpha1 and v1, but it will only **serve** v1. Deploying manifests won't be affected, but code interacting with the operator needs to be prepared to handle v1 objects as response.

This means we need to update all our testsuite, and that the serial suite is no longer backward compatible down to `release-4.10` branch.

All the above is very very similar to what we did in https://github.com/openshift-kni/performance-addon-operators/pull/305

depends on #504

### CODE REVIEW

**Comment by ffromani** [comment]
/cherry-pick release-4.12
not with high hopes, but still

File: doc/examples/nrop.ocpci.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: nodetopology.openshift.io/v1alpha1
```
Comment (swatisehgal) [inline doc/examples/nrop.ocpci.yaml:1]: I understand that `v1alpha1` apiversion would be converted to `v1` apiversion so this is not wrong but shouldn't we update this to the latest supported version i.e. `v1` ? I guess the same applies to the `nrop.yaml `manifest below.

**Review by swatisehgal — COMMENTED** [review]
Kudos! Thanks for your work on this! 

I left a couple of questions mostly for my learning and better understanding. I suggest combining the commits related to config normalization (which is non-blocking) but I think it would make the changes more comprehensible. If you prefer to separate them into separate commits, we should perhaps put them one after the other for the sake of continuity.

File: api/numaresourcesoperator/v1/numaresourcesoperator_conversion.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
```
[the remark this answers is not in the document]
Comment (ffromani) [reply api/numaresourcesoperator/v1/numaresourcesoperator_conversion.go:1]: Correct. I'm following the nice tutorial here which is a great way to get a quick (but still good) grasp of these concepts: https://book.kubebuilder.io/multiversion-tutorial/tutorial.html

File: controllers/numaresourcesoperator_controller.go [hunk 3]
```diff
@@ -144,7 +143,10 @@ func (r *NUMAResourcesOperatorReconciler) Reconcile(ctx context.Context, req ctr
 		return r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, err.Error())
 	}
 
-	normalize.NodeGroupTreesConfig(trees)
+	for idx := range trees {
+		conf := trees[idx].NodeGroup.NormalizeConfig()
+		trees[idx].NodeGroup.Config = &conf
+	}
```
[the remark this answers is not in the document]
Comment (ffromani) [reply controllers/numaresourcesoperator_controller.go:149]: not really, I can merge them, it's no bother (and probably a bit better indeed)

File: pkg/normalize/normalize.go [hunk 4]
```diff
@@ -1,52 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (ffromani) [reply pkg/normalize/normalize.go:1]: sure thing

File: api/numaresourcesoperator/v1/numaresourcesoperator_conversion.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (swatisehgal) [reply api/numaresourcesoperator/v1/numaresourcesoperator_conversion.go:1]: Thanks!

[the remark this answers is not in the document]
Comment (ffromani) [reply api/numaresourcesoperator/v1/numaresourcesoperator_conversion.go:1]: NP!

File: doc/examples/nrop.yaml [hunk 5]
```diff
@@ -4,6 +4,8 @@ metadata:
   name: numaresourcesoperator
 spec:
   nodeGroups:
-  - machineConfigPoolSelector:
+  - config:
```
[the remark this answers is not in the document]
Comment (ffromani) [reply doc/examples/nrop.yaml:7]: nope, we are mistakenly not updating it :) fixing

**Review by swatisehgal — APPROVED** [review]
/lgtm
/approve

Thanks for addressing all the review comments!

**Comment by ffromani** [comment]
> /lgtm /approve
> 
> Thanks for addressing all the review comments!

thank you for the review!

**Pull request merged**